aten/src/ATen/native/mkldnn/Gelu.cpp

```diff
@@ -0,0 +1,89 @@
+#include <ATen/ATen.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/Config.h>
+
+
+#if !AT_MKLDNN_ENABLED()
+
+namespace at { namespace native {
+
+Tensor mkldnn_gelu(const Tensor& input) {
+  TORCH_CHECK(false, "mkldnn_gelu: ATen not compiled with MKLDNN support");
+}
+
+Tensor mkldnn_gelu_backward(const Tensor& grad_output, const Tensor& input) {
+  TORCH_CHECK(false, "mkldnn_gelu_backward: ATen not compiled with MKLDNN support");
+}
+
+}}
+
+#else // AT_MKLDNN_EBABLED
+
+#include <ATen/native/mkldnn/MKLDNNCommon.h>
+#include <ATen/native/mkldnn/Utils.h>
+#include <ATen/native/mkldnn/Gelu.h>
+
+namespace at { namespace native {
+
+Tensor _mkldnn_gelu(const Tensor& input, const Tensor& result) {
+  if (input.scalar_type() == ScalarType::BFloat16) {
+    TORCH_CHECK(mkldnn_bf16_device_check(),
+        "mkldnn_gelu: bf16 path needs the cpu support avx512bw, avx512vl and avx512dq");
+  }
+  bool is_scalar_input = (input.dim() == 0) && (input.numel() == 1);
+  if (is_scalar_input) {
+    input.resize_({1});
```

Comment:
BAD BAD BAD, you're NOT allowed to resize inputs

Reply:
Scalar tensor will go into the original slow path now.